shark_turbine/kernel/wave/thread_shape_analysis.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from ...support.logging import get_logger
+from shark_turbine.kernel._support.tracing import CapturedTrace
+import torch.fx as fx
+from ..ops.wave_ops import *
+from ..lang.global_symbols import *
+from .utils import capture_forward_slice, capture_backward_slice
+
+logger = get_logger("turbine.wave.thread_shape_analysis")
+
+
+@dataclass(order=True)
+class IndexSize:
+    index: IndexSymbol
```

Comment:
minor but to avoid the confusion with index (which is the index sequence in the rest of the code), I would call this dim since it is an IndexSymbol. Then some of the other lines get easier to understand like
```
custom.index[target.dim].size = target.size
```
I think you might want to rename other functions from index -> dim.

Reply:
Done!